xdsl/printer.py

```diff
@@ -119,6 +119,7 @@ def print(self, *argv: Any) -> None:
                 self.print_string(arg)
                 continue
             if isinstance(arg, SSAValue):
+                arg = cast(SSAValue[Attribute], arg)
```

Comment:
This is yucky but tbf this whole function is yucky

Reply:
Yeah, the problem is that `isinstance(arg, SSAValue)` returns an `SSAValue[Unknown]`, and we cannot use `isa` in this file if I recall

Reply:
I'm not sure I fully grasped why pyright can't realise the type is `SSAValue[Attribute]` here using the lower bound on the type variable.

Reply:
https://github.com/microsoft/pyright/discussions/3891

It *could* work if the field is frozen, but would require too many changes in pyright AFAIU

Reply:
Do default typevar types work here?

Reply:
I think it already defaults to `Attribute`, no?

Reply:
It doesn't in this case.
The default only works when doing type checking, here `isinstance(arg, SSAValue)` doesn't use the default as `SSAValue` is a runtime value.
So here, `arg` is a `SSAValue[Unknown]`, which is why I casted it.
Sasha found a workaround, but I believe it is a pyright error, I'm reporting it for now.

I would rather keep the cast for now, while reporting the bug on pyright?

Reply:
Reported here: https://github.com/microsoft/pyright/issues/10214

Reply:
Yeah let's keep the cast